src/fetchBaseQuery.ts

```diff
@@ -110,7 +110,7 @@ export function fetchBaseQuery({
       config.headers.set('content-type', 'application/json');
     }
 
-    if (body && isPlainObject(body) && isJsonContentType(config.headers)) {
+    if (body && isJsonContentType(config.headers)) {
```

Comment:
In all the cases mentioned on https://github.github.io/fetch/#request-body we explicitly do not want to JSON.stringify, but keep the original value, since fetch handles those natively.
Removing the `isPlainObject` check would also stringify those - maybe check if `Array.isArray` OR `isPlainObject` apply here instead?

Reply:
No, because to keep the original value, it looks like that we also need to set the correct `Content-Type` header. By the way, seems that the popular [ky](https://github.com/sindresorhus/ky) is doing this way (JSON.stringify), I double-checked before submitting this PR :

https://github.com/sindresorhus/ky/blob/main/index.js#L247

```js
this._options.body = JSON.stringify(this._options.json);

Reply:
Sorry it took so long to respond, I'm quite busy atm.

I'm not 100% in agreement here - JSON.stringifying a `FormData` will give you `{}` regardless of the contents for example, so that's pretty much never what you want.

Also, this is the code of axios: https://github.com/axios/axios/blob/master/lib/defaults.js#L31 - so the way of `ky` doing it is not really the "default way" to handle this (given that axios has 60x the downloads of ky)



So maybe we should not use the content-type of `application/json` per default, but only for plain objects/arrays and leave it untouched otherwise?

@msutkowski if you have the time - do you have an opinion on this?

Reply:
Sorry about the forever delay here. Looks like the `isPlainObject` was an oversight on my end - here's what I propose:

```ts
    const isJsonifiable = (body: any) => body && typeof body === 'object' && typeof body.append !== 'function';

    // Only set the content-type to json if there is an object that is not FormData()
    if (!config.headers.has('content-type') && isJsonifiable(body)) {
      config.headers.set('content-type', 'application/json');
    }

    if (isJsonifiable(body) && isJsonContentType(config.headers)) {
      config.body = JSON.stringify(body);
    }
```

Reply:
Alright, @phryneas is right here that we don't do thorough checking, but that isn't really the goal of `fetchBaseQuery` _(imo)_. My thought process is being that we're setting the content-type automatically in some cases, it's 'safe-ish' to do what I'm proposing. If someone wanted to opt-out of that, they'd most likely be setting headers on the request anyways, which would avoid this problem. 

@gfortaine , do you have any thoughts there?

Reply:
My alternative suggestion to @msutkowski's `isJsonifiable`:

```js
 const isJsonifiable = (body: any) => typeof body === 'object' && (isPlainObject(body) || Array.isArray(body) || typeof body.toJSON === 'function');

```

Reply:
This would be safer :)